### ISSUE TITLE
Same Twig code for SEO Image Custom URL works differently between Content SEO and Override field

### ISSUE THREAD

**Issue by jmcgaha** [nystudio107/craft-seomatic#1227]
### Describe the bug
I am trying to set fallbacks for images where if the feature Image field is not filled out, pull in the hero background image. This is the code I am setting in the Content SEO area of seomatic for my section.
`{{ seomatic.helper.socialTransform(entry.featuredImage[0] ??? entry.hero[0].backgroundImage[0], "base", 1, "crop") }}`

I cannot get that to output any image. However I do have an SEOMatic Field on that section and if I turn on the override and paste the exact same code it will work.

I have also edited the entry to have both images populated, the asset field not populated, the hero matrix with no rows then with no image and the behavior is always the same: Override version always works, the Content SEO version never works.

I feel this is similar to the ticket #1078 but the solution of trying the twig on a page is not a valid response since it does work in the override field.

### To reproduce
Steps to reproduce the behaviour:
1. Set a custom URL for SEO Image in the content SEO area with a fallback query.
2. See if that works on an entry.
3. If it does not, try the same query in the override field.

### Expected behaviour
The specified query should work when used in the Content SEO area without having to specify an override.

### Versions
- Plugin version: 3.4.40
- Craft version: 3.7.34

<img width="1539" alt="Content SEO" src="https://user-images.githubusercontent.com/49672/198330981-a3cd7294-d864-4e6f-907a-91d37b991637.png">
<img width="1617" alt="Override on" src="https://user-images.githubusercontent.com/49672/198330950-a3d4845f-e803-43e1-88e6-e4f71b81c795.png">
<img width="1616" alt="Override off" src="https://user-images.githubusercontent.com/49672/198330973-a388df2c-e50f-42ed-aa74-de77331f4ab9.png">

**Comment by khalwat** [comment]
If you put this code:
```twig
{{ seomatic.helper.socialTransform(entry.featuredImage[0] ??? entry.hero[0].backgroundImage[0], "base", 1, "crop") }}
```

...in your frontend entry template for the page in question, what does it output?

**Comment by jmcgaha** [comment]
- When both fields are filled out that outputs the transformed image url of the Featured Image.
- When just Featured Image is filled out it outputs the transformed image url of the Featured Image.
- When just the Hero Background Image is filled out it outputs the transformed image url of the Hero Background Image.
- With neither field filled out and the matrix block being empty with no blocks/rows, it gives an error about the matrix block not existing.

**Comment by khalwat** [comment]
I'm going to try again to reproduce this tomorrow; I'm a bit perplexed as to why I've been unable to thus far.

**Comment by khalwat** [comment]
So I replicated your setup as best I can understand from the issue. From what I can see, you have an Entry that has:

* An Asset field with the handle `featuredImage`
* A Matrix field named `hero`
* A field in the `hero` Matrix field named `backgroundImage`

Breaking down the Twig...

 `entry.featuredImage` will either be:
* An `AssetQuery` is an image is set, but not eager loaded
* An array if an image is set, and they _are_ eager loaded

In this case, the null coalescing operator will do the right thing if it's null, 

`entry.hero` will either be:
* `null` if there are no matrix blocks
* A `MatrixBlockQuery` if there are matrix blocks, but not eager loaded
* An array if there are matrix blocks, and they _are_ eager loaded

The problem is, if no matrix blocks exist, `entry.hero` returns an empty array, and you can't access the `entry.hero[0]` if the array is empty.

The output from putting that code into your template is correct, it should throw an error if no matrix blocks exist, but the other 3 cases work as they should. I also tested this same code entered into the Content SEO Settings in SEOmatic, and it also works as expected.

So the problem here is that the Twig code isn't correct, in terms of what you want to do with it. The only reason it works in the SEO Settings field is it catches the error, and doesn't render anything in that case.

Probably you'll want to write something like this: 

```twig
    {{ seomatic.helper.socialTransform(entry.featuredImage[0] ??? entry.hero.one() | length ? entry.hero.one().backgroundImage[0] : '' ??? '', "base", 1, "crop") }}
```
So if `entry.hero.one()` returns some matrix blocks, use `entry.hero.one().backgroundImage[0] ` otherwise, return an empty string.

The additional `??? ''` at the end is added to catch the case where `entry.hero.one().backgroundImage[0]` is empty, so it defaults to an empty string. Which will cause it to fall back on the global SEO settings if nothing is set.

**Comment by jmcgaha** [comment]
Thanks for this explanation. When you first posted the working code in November I tried it but still didn't have any success, the SEO image on the entry was always blank. I left it alone until I had a chance to loop back today and I finally figured it out.

I have been making these changes in the Content SEO area, which is where they should be. But I just had a hunch that maybe I should just re-save the settings in the Global SEO area. When I did, still no dice.

The Global SEO -> SEO Image Source was set to a "Custom Image" that was blank. If I changed that to a blank "Custom URL", now the conditionals I set in the Content SEO area are working, if I change it back to "Custom Image" the Content SEO settings no longer work.

I'm not sure this is a bug, but I don't think they should necessarily be related. Either way just wanted to post my fix in case anyone else runs into this.
